stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+import logging
+
+from odoo import api, fields, models
+from odoo.tools.safe_eval import safe_eval
+
+logger = logging.getLogger(__name__)
+
+
+class StockWarehouseFlow(models.Model):
+    _name = "stock.warehouse.flow"
+    _description = "Stock Warehouse Routing Flow"
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True)
+    warehouse_id = fields.Many2one(
+        comodel_name="stock.warehouse",
+        ondelete="restrict",
+        string="Warehouse",
+        default=lambda o: o._default_warehouse_id(),
+        required=True,
+    )
+    from_picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type",
+        ondelete="restrict",
+        string="From operation type",
+        required=True,
+    )
+    from_location_src_id = fields.Many2one(
+        comodel_name="stock.location",
+        compute="_compute_from_location_src_id",
+    )
+    from_location_dest_id = fields.Many2one(
+        comodel_name="stock.location",
+        compute="_compute_from_location_dest_id",
+    )
+    to_picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type",
+        ondelete="restrict",
+        string="To operation type",
+        required=True,
+        domain="[('default_location_dest_id', '=', from_location_dest_id)]",
```

Comment:
We need a to_location_src_id, could be a child of the from_location_src_id. This is in order to be able to pull the rule of the flow (and not the default one from the WH, as the procurement is searching a rule according to the location_src)
Then the picking type source location has to be that to_location_src_id. Ideally, we should be able to generate that picking type as you generate the route.

Reply:
Operation types are now automatically generated.